extension/src/format/sexpression.ts

```diff
@@ -817,52 +1080,56 @@ export class Sexpression extends LispAtom {
     //    And we treat the block comments as a common atom
     //
     // All the atom index start from 0, and include the left parenthes
-    public format(startColumn: number, asCond?: boolean): string {
-        let length = this.length();
-        if (this.isDotPairs()) {
-            return this.formatAsPlainStyle(startColumn);
-        }
-        else if (this.isQuote()) {
-            return this.formatQuote(startColumn);
-        }
-        else if (length > 4) {
-            let lispOperator = this.getLispOperator();
-            if (this.atoms[0].isLeftParen() &&
-                !this.canBeFormatAsPlain(startColumn)) {
-
-                let opName = lispOperator.symbol.toLowerCase();
-
-                if (opName == "if" || opName == "repeat" || opName == "while")
-                    return this.formatList(startColumn, 3);
-                if (opName == "lambda")
-                    return this.formatLambda(startColumn);
-                else if (opName == "cond")
-                    return this.formatList(startColumn, 2, true);
-                else if (opName == "setq") {
-                    return this.formatSetq(startColumn);
-                }
-                else if (opName == "foreach") {
-                    return this.formatForeach(startColumn);
-                }
-                else if (opName == "defun" || opName == "defun-q") {
-                    return this.formatDefun(startColumn);
-                } else if (this.isPureLongList() && gLongListFormatAsSingleColumn == LongListFmts.kFitToMargin)
-                    return this.formatListAsColumn(startColumn, 3);
-
-                if (asCond) {
-                    // cond branch internal expression align with the outer left parenthes
-                    return this.formatList(startColumn, 1);
-                }
-
-                if (this.shouldFormatWideStyle(startColumn))
-                    return this.formatListAsWideStyle(startColumn);
-                else {
-                    return this.formatListAsNarrowStyle(startColumn);
+    static format(exp: LispAtom|Sexpression, startColumn: number, asCond?: boolean): string {
```

Comment:
I mostly just made this static, changed the argument signature to take the object reference and replaced `this` with `exp` everywhere. I didn't even go add brackets around the various IF statements that are complaining; seriously touched as little as possible.